src/NServiceBus.Core/Transports/Sending.cs

```diff
@@ -1,53 +1,60 @@
+#pragma warning disable 1591
 namespace NServiceBus
 {
-    using System;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using Features;
+    using ObjectBuilder;
+    using Routing;
+    using Settings;
     using Transport;
 
-    class Sending : Feature
+    public class TransportComponent
     {
-        public Sending()
+        public TransportComponent(TransportInfrastructure transportInfrastructure)
         {
-            EnableByDefault();
+            TransportInfrastructure = transportInfrastructure;
         }
 
-        protected internal override void Setup(FeatureConfigurationContext context)
-        {
-            var transport = context.Settings.Get<OutboundTransport>();
-            var lazySendingConfigResult = new Lazy<TransportSendInfrastructure>(() => transport.Configure(context.Settings), LazyThreadSafetyMode.ExecutionAndPublication);
-            context.Container.ConfigureComponent(c =>
-            {
-                var dispatcher = lazySendingConfigResult.Value.DispatcherFactory();
-                return dispatcher;
-            }, DependencyLifecycle.SingleInstance);
+        public IDispatchMessages Dispatcher { get; set; }
```

Comment:
I have some doubts about exposing this one as a plain object here. I'd rather have some safety net that ensures it is not invoked until after initialize. 

I am thinking about wrapping it in a decorator that ensures we are at least in a starting phase. Thoughts?